springboot/src/main/java/ca/jrvs/apps/trading/dao/MarketDataDao.java

```diff
@@ -0,0 +1,209 @@
+import org.springframework.data.repository.CrudRepository;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class MarketDataDao implements CrudRepository<IexQuote, String> {
+    private static final String IEX_BATCH_PATH = "stock/market/batch?symbols=%s&types=quote&token=";
+    private static final int HTTP_OK = 200;
+    private static String IEX_BATCH_URL;
+    private Logger logger = LoggerFactory.getLogger(MarketDataDao.class);
+    private HttpClientConnectionManager httpClientConnectionManager;
+
+    @Autowired
+    public MarketDataDao(HttpClientConnectionManager httpClientConnectionManager, MarketDataConfig marketDataConfig) {
+        this.httpClientConnectionManager = httpClientConnectionManager;
+        IEX_BATCH_URL = marketDataConfig.getHost() + IEX_BATCH_PATH + marketDataConfig.getToken();
+    }
+
+
+    @Override
+    public <S extends IexQuote> S save(S s) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public <S extends IexQuote> Iterable<S> saveAll(Iterable<S> iterable) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    /**
+     * Get an IexQuote (Helper method which class findAllById)
+     *
+     * @param ticker
+     * @throws IllegalArgumentException      if a given ticker is invalid
+     * @throws DataRetrievalFailureException if HTTP request failed
+     */
+    @Override
+    public Optional<IexQuote> findById(String ticker) {
+        Optional<IexQuote> iexQuote;
+        List<IexQuote> quotes = findAllById(Collections.singletonList(ticker));
+
+        if (quotes.size() == 0) {
+            return Optional.empty();
+        } else if (quotes.size() == 1) {
+            iexQuote = Optional.of(quotes.get(0));
+        } else {
+            throw new DataRetrievalFailureException("Unexpected number of quotes");
+        }
+        return iexQuote;
+
+    }
+
+    @Override
+    public boolean existsById(String s) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public Iterable<IexQuote> findAll() {
+        return null;
+    }
+
+    /**
+     * Get quotes from Iex
+     *
+     * @param tickerList is a list of tickers
+     * @return a list of IEX quote object
+     * @throws IllegalArgumentException      if any ticker is invalid or ticker is empty
+     * @throws DataRetrievalFailureException if HTTP request is failed
+     */
+    @Override
+    public List<IexQuote> findAllById(Iterable<String> tickerList) throws IllegalArgumentException,
+            DataRetrievalFailureException {
+        int tickerNumber = 0;
+        for (String ticker : tickerList) {
+            if (!ticker.matches("[a-zA-Z]{2,4}")) {
+                throw new IllegalArgumentException("Illegal ticker format!");
```

Comment:
as per javadoc, you are supposed through `DataRetrievalFailureException` exception.